gm4_blossoming_pots/data/gm4_blossoming_pots/function/flower/raycast/in_pot.mcfunction

```diff
@@ -0,0 +1,20 @@
+# checks if execution context is inside the flower pots hit box
+# @s = player who interacted with a flower pot
+# at @s anchored eyes, then marched forward 0.005 over and over. Complicated.
+# run from flower/raycast/loop
+
+# early return if not flower pot
+execute unless block ~ ~ ~ minecraft:flower_pot run return fail
+
+# summon marker for checking
+summon minecraft:marker ~ ~ ~ {Tags:["gm4_blossoming_pots.temp.flower_pot","smithed.strict","smithed.entity"],CustomName:'{"text":"gm4_blossoming_pots.temp_marker"}',data:{id:null}}
```

Comment:
What is this `id` field?

Reply:
The temp marker summoned here has the players SelectedItem stored on it, as well as the players UUID, and rotation. This happens in flower/raycast/player_rotation. id is set to null here as the fall back value for if the SelectedItem of the player is null and doesn't write. This is to distinguish an empty hand right click.